### DISCUSSION TITLE
Does MAUI support win7?

### DISCUSSION THREAD

**Discussion by heartacker** [dotnet/maui#2576] [Q&A]
![image](https://user-images.githubusercontent.com/1876302/133871405-a1263a20-8d0a-487a-9abe-252e162a3a56.png)

please tell me, thanks

**Comment by UweReisewitz** [comment]
WinUI3 requires at least Windows 10 1809. It does not support Win7, earlier versions of Windows 10 or Server versions before Server 2019

**Reply by CommonLoon102** [reply]
MAUI Linux (https://github.com/jsuarezruiz/maui-linux) is using GTK3 to run MAUI apps on Linux and GTK3 is available for Windows 7 as well. So in theory it would be possible to have Windows 7 support, if someone is willing to do the effort.

**Reply by saint4eva** [reply]
why not use Uno.Platform ?

**Reply by CommonLoon102** [reply]
UNO is using XAML, while in MAUI you can use web technologies (Blazor components, HTML with razor, CSS, bootstrap, etc.).

**Reply by saint4eva** [reply]
Yes, you are right, Uno uses xaml. And I think that is a downside for me also. Although they are planning to have MVU C# UI for writing UI. However, Uno is so powerful that it can accomplish anything you could do with CSS, bootstrap html etc.

**Reply by CommonLoon102** [reply]
And just a simple (or not so simple) calculator consumes 500 MB memory. Selecting text is also problematic.

And this page:
https://gallery.platform.uno/
consumes 1.2-2.2 GB memory, depending on how much you click around.

**Reply by sogeisetsu** [reply]
> why not use Uno.Platform ?

@saint4eva . uno is nice. It supports almost all platforms, but its development community is always difficult to be reassuring. It is not known by many people now. I know that it is also on the official website of winui. On GitHub, there are nearly 1,000 issues that have not been closed and quite a few of them have not been answered. It is awesome, but there is no ecology. It is not like a great company like Microsoft behind maui.

I think we need to spend a little time watching uno. Now out of caution, I still pin my hopes on Microsoft. Windows 7 still has 27% of users in the world. As a developer, I can't give up so many users, but I am not reconciled to only use technologies like wpf that are not a few years younger than me.

**Reply by GeraudFabien** [reply]
> Windows 7 still has 27% 

Most of windows 7 user are in china
And if you remove china most are ;
- secure environment where you can't do any thinks unless you're SI allowed it.
- PC use in a really specific case without internet and only one application install on it
- Test PC for developer that need to be support

Except if you target one of this three case you don't need windows 7 support. Also remember that it's hard to know the market share. Event MS don't know how many PC use Win7 so most of this number are from site or use method that are far from being accurate. Some estimate are at 50% and some are as less than 10%. 

> uno is nice. It supports almost all platforms, but its development community is always difficult to be reassuring.

Has for uno it's support by a growing community. Microsoft remove support of WinForm, WPF, UWP (WIN UI 2), WebForm, WCF, Silverlight, XNA. I know that the support of all except XNA and Silverlight are still officially support but there is nearly no support.

![image](https://user-images.githubusercontent.com/13610689/140606546-5b08c95d-295c-40a2-83df-d76ebf24e6b6.png)

**Reply by CommonLoon102** [reply]
> * secure environment where you can't do any thinks unless you're SI allowed it.
> * PC use in a really specific case without internet and only one application install on it
> * Test PC for developer that need to be support

I would also add gamers. A lot of older games don't work on newer Windows versions.
Based on the Steam Hardware & Software Survey, the second most used OS is Windows 7 64 bit as of 2021 October.
More people use Win7 for gaming than Linux and Mac altogether.
https://store.steampowered.com/hwsurvey

And there are a lot of other people who just don't like Win8, Win10, Win11 for various reasons, and won't ever upgrade from Win7.

**Reply by sogeisetsu** [reply]
> And if you remove china 

but I can not remove china because I come from china. In china there are so many people use windows7. I guess the number of people using win7 in China may exceed 100 million (there are nearly 1.4 billion people in China). 🤣.And because the Chinese government prohibits the government from purchasing windows10, it is possible that almost all Chinese users who use windows10 now buy windows10 on their own or use windows10 gifted by laptops. Most people who work in the Chinese government use windows7 or xp in the office. I even see it with my own eyes. I saw that the traffic police office uses Windows 98.😫.I believe that the number of people using Windows 7 in the world is definitely not a small number.

>  I know that the support of all except XNA and Silverlight are still officially support but there is nearly no support.

The future of uno is limitless, uno is developing, but WPF seems to have stopped developing, but WPF has been developed for many years, and Microsoft's support for WPF a few years ago is very strong, after many years, it has become better. But uno is a GitHub repository created on 2018-05-07 11:52:27. I am looking forward to uno, but I don't think that uno is more stable than those frameworks you mentioned.

**Reply by CommonLoon102** [reply]
> but I can not remove china because I come from china. 

But how can you even open GitHub from China? Isn't it blocked by the GFW?

**Reply by sogeisetsu** [reply]
> > but I can not remove china because I come from china.
> 
> But how can you even open GitHub from China? Isn't it blocked by the GFW?

I don't know why you think you can't access GitHub from China.🤷‍♂️

I believe you are not mocking me, but you do have questions about "GFW". 😅Maybe this is a stereotype. Not all online activities are banned in China. The government will only ban websites that are suspected of insulting the country. At least official websites like Amazon, GitHub, Apple and Microsoft are all accessible normally. We will also use Google Translate and gmail. Each country has its own national conditions, and perhaps there are things in each country that cannot be mentioned, just like a former president(Trump) of the United States would be blocked from Twitter for inappropriate remarks.🤣

During our National Day, we will indeed be unable to access GitHub for a few days. But as long as you submit a reasonable application, you can also be allowed to use GitHub. Because some people are inseparable from GitHub for their normal work.😁

I don't want to discuss politics or care about politics, I just want to discuss MAUI here. (I say this because I am afraid that someone will quarrel here because of China, which is very common on Twitter.)🤦‍♂️🤣

**Reply by CommonLoon102** [reply]
> I don't know why you think you can't access GitHub from China.man_shrugging

I thought that GitHub was banned after the 996.icu repo.

**Reply by saint4eva** [reply]
> > why not use Uno.Platform ?
> 
> @saint4eva . uno is nice. It supports almost all platforms, but its development community is always difficult to be reassuring. It is not known by many people now. I know that it is also on the official website of winui. On GitHub, there are nearly 1,000 issues that have not been closed and quite a few of them have not been answered. It is awesome, but there is no ecology. It is not like a great company like Microsoft behind maui.
> 
> I think we need to spend a little time watching uno. Now out of caution, I still pin my hopes on Microsoft. Windows 7 still has 27% of users in the world. As a developer, I can't give up so many users, but I am not reconciled to only use technologies like wpf that are not a few years younger than me.

Nventive is behind Uno, and Nventive is a big company - a Canadian company. And many big companies are using Uno as their preffered tech solution. Kindly note that Microsoft built its ecosystem in such a way that third-party solutions and vendors are mostly welcome. Uno is more than viable solution.

https://nventive.com/en/

WPF and WinForm, including Uno are great solutions for Windows 7. Mind you, .NET Maui targets Windows ecosystem by using WinUI. So, if WinUI does not support Windows 7, Maui would not do anything.

**Reply by softboy99** [reply]
winui comes from previous UWP. However, Why UWP was not popular? because it not considering compatibility, it only support win10. MAUI will be next UWP. If you don't believe it, wait and see.

**Comment by nextcodelab** [comment]
> ![image](https://user-images.githubusercontent.com/1876302/133871405-a1263a20-8d0a-487a-9abe-252e162a3a56.png)
> 
> please tell me, thanks

I think It should not support Windows 7 to make the Windows 11 the default choice for security reasons. And it is also win for the developers and consumers since it easy to create apps in windows 11/windows 10 than windows 7. Windows 7 is an old looking design computer like symbian OS. Windows 7 is like 1950's computer. Look mostly programs in windows 7 are super old, classic and heavy no smooth apps. Sorry for my opinions that seems direct.

I just hoping that all games in windows 7 will run smoothly on windows 11 with just additional frameworks to be installed, that's maybe an issue if I'm not mistaken.

**Reply by CommonLoon102** [reply]
> I just hoping that all games in windows 7 will run smoothly on windows 11 with just additional frameworks to be installed, that's maybe an issue if I'm not mistaken.

Some games don't even work on Windows 7 what were working on Windows XP (e.g. Caesar 4). For that there is Wine. Windows is less compatible with itself than Wine with Windows in some cases.

**Reply by davidbuckleyni** [reply]
The answer has already been giving no plz close this thread @PureWeen or someone

**Comment by sogeisetsu** [comment]
@heartacker maybe you should close the discussion because there are quit a few anwsers here. MAUI dose not support windows 7 and mircosoft did not tell the dev that will have any plan to make MAUI support win7

**Reply by saint4eva** [reply]
Maui will not support Windows 7, because the target for Windows platform is WinUI. And WinUI does not have any plans to support Windows 7. However, if anyone wants to develop for Windows 7, the person can use Uno.Platform. Uno is a great tool and it is supported by a big Canadian company.

**Comment by hanjunjun** [comment]
We also have a large number of C-end products that use win7. If win7 is not supported across platforms, we can only replace other cross-platform client solutions.For example, uno.

**Comment by davidbuckleyni** [comment]
This will not happen folks MS has already said that even windows 10 end of life will happen soon to expect them to keep supporting  windows 7 is crazy. You can stay on visual studio versions prior to 2022 allot of orgs have done this for a long time

**Comment by softboy99** [comment]
stupid decision.

**Comment by yyliveyy** [comment]
WIN7 is dead and support for it is meaningless.

**Comment by softboy99** [comment]
.net will never replace java, if you never open your mind

**Reply by CommonLoon102** [reply]
Java on the client is even more dead than Windows 7.

**Reply by softboy99** [reply]
"Microsoft support for Windows 7 started to end in 2020. You can develop Apps in WPF that are more powerful than what MAUI offers at this time, still reuse 85% of your business logic code and WPF is a great XAML learning tool for what MAUI can become."

Please also deny this for windows 7!!!!!! So that its really cross any platform!

**Comment by a44281071** [comment]
no win7, no maui.

**Reply by saint4eva** [reply]
Move on mate

**Reply by CommonLoon102** [reply]
Chromium won't support Win 7 after Jan 2023. Only a few months left. Then anything electron based won't work anymore on Win 7 either.

**Reply by softboy99** [reply]
"Microsoft support for Windows 7 started to end in 2020. You can develop Apps in WPF that are more powerful than what MAUI offers at this time, still reuse 85% of your business logic code and WPF is a great XAML learning tool for what MAUI can become."

Please also deny WPF for windows 7!!!!!! So that its really cross any platform!

**Reply by softboy99** [reply]
Unitil now ,VSCODE still based on chromium 100, at least it will be 2024. Not every electron based app will update to latest chromium immediatly

**Reply by CommonLoon102** [reply]
> Unitil now ,VSCODE still based on chromium 100, at least it will be 2024. Not every electron based app will update to latest chromium immediatly

True, but sooner or later it is going to be a problem for Win7 users.

**Reply by softboy99** [reply]
why more and more developers leave .net？ They are always abandoned. Microsoft are the greedy dog bear, which always picks up a stick and lost his own stick.

**Reply by saint4eva** [reply]
> why more and more developers leave .net？ They are always abandoned. Microsoft are the greedy dog bear, which always picks up a stick and lost his own stick.

Can we have source of your statistics?

**Reply by a44281071** [reply]
Yes, cross platform (Just win10/11).

**Reply by CommonLoon102** [reply]
> Yes, cross platform (Just win10/11).

And Mac and iPhone and Android. That's already 3 other besides Windows, so it is cross-platform indeed.

**Reply by softboy99** [reply]
> And Mac and iPhone and Android. That's already 3 other besides Windows, so it is cross-platform indeed.

indeed, it's not

**Reply by softboy99** [reply]
Microsoft is the greedy dog bear, picks up a stick and throw away another. WPF, WWF,WCF ,Silverlight........., You abandon the developers. When your modernize your tech, doesn't consider smoothly transition any more, Which is your strong point before. Win the tech, lose the heart, lose the market. So big joke, the most popular tech currently is still winform(win32)

**Reply by MicroTrendsTom** [reply]
yes bring back locomotive basic.... all this exciting progress with and better technology is terrible, i'm still on win3.11  hymem.sys, command.com  ;-)      Win 7 is not supported,.. get over it folks...   

WPF was the first iteration of a good idea, in reality it was a bloated poor horrific implementation for a long time...  
WinUI and Maui are the next evolution embrace it -  MAUI blazor... oh love it.

**Reply by softboy99** [reply]
The acquisition of Xamarin didn't inherit the original intention of Xamarin. Xamain support win7, MAUI merged Xamrin but lost many. So it's not a tech issue. It's a fuck business issue.

**Comment by softboy99** [comment]
the greedy dog bear always picks up a stick and lost his own stick

**Reply by softboy99** [reply]
support win7 is not an techniqual issue, but, a fuck business issue

**Reply by CommonLoon102** [reply]
Win7 is not supported by MS since more than 2 years, is this something new for you?
You have to use VS2022 latest to get MAUI support and that is only compatible with Win10/11. So how would you develop if you have Win7?

Why don't you use Linux with GTK or Qt instead?
Or if you want to use XAML, there is platform.uno. There are so many alternatives.

**Reply by sdhongjun** [reply]
So what is MAUI's aim? Lets developer choose other cross platform solution.

**Reply by softboy99** [reply]
@CommonLoon102，VS2022 is not an an techniqual issue too,it's build with .net

**Reply by CommonLoon102** [reply]
@softboy99 
More and more features won't work with VS2022 on Windows 7.
The tooling support is going to end sooner or later. There are issues already:
https://github.com/dotnet/aspnetcore/issues/38188
The issue is "Resolved" but I reported it with the internal feedback tool and there the engineers said that it is some hickup between the browser and VS2022 and only Windows 7 is affected but they won't fix it since they cannot reproduce because they don't have Windows 7 because it is not even supported.

I had issues with C++ projects as well, I can't compile them with VS2022 anymore (only with VS2019).

**Reply by softboy99** [reply]
windows7 also can install edge browser instead of IE

**Reply by CommonLoon102** [reply]
> windows7 also can install edge browser instead of IE

Blazor hot reload still won't work with it, only on Windows 10, I tried it.
In MAUI you can have a Blazor WebView and you won't be able to test it because hot reload is broken on Windows 7, it is not supported.

BTW you won't be able to update Edge on Windows 7 after January 2022. You will stuck on an old version.

**Reply by softboy99** [reply]
without edge, we have chrome

**Reply by CommonLoon102** [reply]
> without edge, we have chrome

No, Chrome has the same end of life for Windows 7. You won't receive Chrome updates after January 2022 on Windows 7. Since Edge is just a fork of Chrome (Chromium actually), it has the same end of life date for Win7. You will still be able to use the latest Firefox though. But I don't know for how long.

**Reply by softboy99** [reply]
@CommonLoon102 ,  both edge & chrome for win7 is supported to 2023.01.15. I think it will be extended due to COVID-19. Your MAUI are long before that

**Reply by CommonLoon102** [reply]
> @CommonLoon102 , both edge & chrome for win7 is supported to 2023.01.15. I think it will be extended due to COVID-19.

Due to COVID-19? I don't think so. What does it have to do with it? In North America and in Europe COVID-19 is not a problem anymore, our vaccines work, no-one gives a damn.

Google has already extended this deadline twice, both times by 1 year. Now it has the same deadline as the paid extended support for Windows 7 (available to companies only, costs a lot of money per computer).

> Your MAUI are long before that

Microsoft won't make their new products and features compatible with an OS what they don't support since years, it would be a waste of time for them. If something new still works, that's just coincidence.

**Reply by softboy99** [reply]
@CommonLoon102,    You can still insist your idea. The dotnet market share ratio in recent years has been proved. You make a tool,named cross problem but cannot cross your own windows. So big joke! It's like JVM, but not equal for ever. If you cannot apart your Business from tech. See Java, There're still support for java8 for many years. Of course, it will be abandon finally. But it win the heart of developers and win the market.

**Reply by CommonLoon102** [reply]
Use Java then.

**Reply by CommonLoon102** [reply]
> @CommonLoon102 , both edge & chrome for win7 is supported to 2023.01.15. I think it will be extended due to COVID-19. Your MAUI are long before that

@softboy99 
https://support.google.com/chrome/thread/185534985/sunsetting-support-for-windows-7-8-1-in-early-2023

**Reply by softboy99** [reply]
> Microsoft won't make their new products and features compatible with an OS what they don't support since years, it would be a waste of time for them. If something new still works, that's just coincidence.

win7 ESU suport is ended after 2023.01.10 by Microsoft. You MAUI goes GA at 2022

**Reply by CommonLoon102** [reply]
Doesn't worth the effort to support something for just 6 months. If it was 6 days, would you still complain? Or 6 seconds?
Actually, most of the people won't start doing anything in MAUI until .NET 7 comes out. It will come in November. So we are talking about 2 months. In December most of the people are on holidays. So it is just a few weeks to support...

**Reply by CommonLoon102** [reply]
I have a 20 years old Athlon XP and I can barely boot it. I had to downclock it from 1.9GHz to 1.1GHz to have it run stable without restarting. If you bought a new PC even 5 years ago, you already couldn't find drivers for Windows 7, especially if that was a laptop. On what machine do you want to run Win7 5 years from now? Because that machine will be around 10-15 years old. CPU's don't live forever, if you use the machine on a daily basis, the transistors will get older and it takes more time for them to switch the gates. Eventually, the CPU will fail if you don't downclock it. How many users are going to do that? So they can run MAUI apps on a 12+ years old machine running on Windows 7.

**Reply by softboy99** [reply]
It's only said that, If not business concern, your tech is still tightly bounded with windows os, there're still many P/Invoke like .net framework. So that stop many people from choosing it.

**Comment by xtuzy** [comment]
If someone can give money to Avalonia team, maybe Avalonia team can create a Maui layer like WinUi3, that mean Maui can support linux, net6.0-mac, win7....

**Comment by mzayarnyuk** [comment]
again bad ms; if support only 1809 why allow poiting 7.0 8.0 and compile; ind noobs

**Comment by mzayarnyuk** [comment]
again needs to create 2 projects to support old devices....

**Comment by giuseppenovielli** [comment]
Simple question, but you can continue to use Xamarin.Forms.WPF -> https://www.nuget.org/packages/Xamarin.Forms.Platform.WPF#readme-body-tab
and ignore MAUI??

Yes not the best... but for small projects....

**Reply by saint4eva** [reply]
If the under lying technology - WinUI 3 - does not support Windows 7, Maui will not support Windows 7

**Reply by maxoptimus** [reply]
Xamarin.Forms for old mobile; uwp for old win10 and xbox; Xamarin.Forms.WPF for win7 and xp;

maui for zero and lose perfomance or at least for new mobile if xamarin stack on some points;

java is apways dead and all android java is clone of .net 2.0 winform - google need more brain all devs known what is on google side android - cloned all 1:1 and added only custom xml parser; goog;e a bigest plagiator after ms on starting;
